extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/conversational.py

```diff
@@ -28,43 +28,37 @@
 # Step 1: define Helpers
 def refine_completion_params(model_settings: dict[Any, Any]) -> dict[str, Any]:
     """
-    Refines the completion params for the HF text translation api. Removes any unsupported params.
-    See https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/inference/_client.py#L1703
-    for supported params. Note that client params DO NOT match API docs here:
-    https://huggingface.co/docs/api-inference/detailed_parameters#translation-task
+    Refines the completion params for the HF conversational api. Removes any unsupported params.
+    Api Ref: https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/inference/_client.py#L338
+    Doc Ref: https://huggingface.co/docs/huggingface_hub/package_reference/inference_client#huggingface_hub.InferenceClient.conversational
     """
 
     supported_keys = {
         "model",
-        "src_lang",
-        "tgt_lang",
     }
 
-    completion_data = {}
+    completion_data: dict[str, Any] = {}
+
     for key in model_settings:
         if key.lower() in supported_keys:
             completion_data[key.lower()] = model_settings[key]
 
-    return completion_data
-
-
-def construct_output(response: str) -> Output:
-    metadata: dict[str, str] = {"raw_response": response}
-
-    output = ExecuteResult(
-        **{
-            "output_type": "execute_result",
-            "data": response,
-            "execution_count": 0,
-            "metadata": metadata,
-        }
-    )
-    return output
+    if "model" in completion_data:
+        # HF is in the process of revamping their libraries. Default behaviour of Converstaional is broken
+        # This temporary fix is reccomended by HF https://github.com/huggingface/huggingface_hub/issues/2023#issuecomment-1935760040
+        completion_data["model"] = (
+            f"https://api-inference.huggingface.co/pipeline/conversational/{completion_data['model']}"
+        )
+    else:
+        # manually set default model because, otherwise this will fail. see comment above.
+        # see this thread for more info: https://github.com/huggingface/huggingface_hub/issues/2023
+        completion_data["model"] = "https://api-inference.huggingface.co/models/facebook/blenderbot-400M-distill"
```

Comment:
nit: Just want to sanity check that this works for free so will be good for most people

Reply:
Unfortunately I am unable to thoroughly test this locally due to unknown rate limiting issues. Will fix forward anything that comes up